hub/experimental/dataloader.py

```diff
@@ -260,4 +258,5 @@ def __iter__(self):
 
 
 def dataloader(dataset) -> Hub3DataLoader:
+    verify_base_storage(dataset)
```

Comment:
verifying base storage using a utility function instead of conversion to hub3. Moreover this check happens only once now, compared to multiple times while chaining earlier.